internal/cli/url_scheme.go

```diff
@@ -0,0 +1,36 @@
+package cli
+
+import (
+	"fmt"
+	"net/url"
+)
+
+type UrlScheme string
+
+const (
+	httpsScheme UrlScheme = "https"
+	httpScheme  UrlScheme = "http"
+)
+
+func addUrlScheme(urlString string, scheme UrlScheme) (string, error) {
+	parsedUrl, err := url.Parse(urlString)
+	// The url.Parse function may return an error when there's no scheme due to parser ambiguities,
+	// so we need to verify if the error is because the url lacks a scheme
+	if err != nil && parsedUrl.Scheme != "" {
+		return "", err
+	}
+
+	if parsedUrl.Scheme == string(scheme) || urlString == "" {
```

Comment:
We might not need the or case because we won't want to return the empty urlString

Reply:
It might be worth explicitly checking this separately in its own case. And if we do want this check to be an ok case, we might want to move it up before the url parsing. In any case, it should match the current behavior.